### PR TITLE
Top navigation, schema selector

### LINKED ISSUE #312: Switch frontend code to the database API
**Problem**

We'd like to remove the database keys API in favor of the new `v0/databases/` API, which can include more information.

**Proposed solution**

Switch frontend code from using the database keys API to the database API.

**Additional context**

- Blocked by #311

### LINKED ISSUE #316: Switch to selected schema in file import view
**Description**
As part of milestone 2(File import), we've hardcoded the schema to be the first available schema. This is because milestone 2 does not include the db selector.

**Expected behavior**
The file import view should take the schema selected on the db/schema selector.

Refer these comments:
* https://github.com/centerofci/mathesar/discussions/305#discussioncomment-943298 
* https://github.com/centerofci/mathesar/discussions/305#discussioncomment-945048

**Additional context**
- Blocked by #362

### LINKED ISSUE #362: Implement new top navigation and layout
## Problem

The [schemas design spec](https://wiki.mathesar.org/en/design/specs/schemas) introduced a new top navigation and layout.

## Proposed solution

 We need to implement the new top navigation and layout. This involves
- Removing schemas from the sidebar.
- Implementing the "Search or jump to..." bar in the top navigation.
- Updating the frontend code and structure so that there's only one active schema at any given time.

### PR DESCRIPTION
**Pull request by pavish**
Fixes #362 
Fixes #316 
Fixes #312 
Related to #173 

This PR:
* New db + schema selector
* Handles redirection for empty/invalid db and schema, schema id is now part of url
* Rewrites schema store to take selected db into account
* New database store, uses database apis
* Components:
  - TextAvatar component
* Adds a static skeleton to indicate record loading
* Adds a dummy favicon to avoid additional favicon requests

Note:
* This PR does not handle closing and reopening of tables, based on selected schema.
  - Created issue https://github.com/centerofci/mathesar/issues/524 to keep track of it.
* Improvises database + schema switching, to avoid rewrites when getting to database switching milestone.
  - It does not follow the UX closely, but since we will have to reconsider the top navigation when get to the database switching milestone, this should be okay.
* This PR does not implement a "Search or jump to..." feature.
   - The menu contains all the schemas, which enables easy switching. The "Search or jump to..." feature needs to be thought out more, since this search bar is essentially a product wide search, and it would require discussions and backend work.

## Screenshots
Screenshot 2021-08-06 at 1 35 06 AM
Screenshot 2021-08-06 at 1 35 22 AM
Screenshot 2021-08-06 at 1 35 44 AM

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/views/frontend.py
Comment (kgodey) [inline mathesar/views/frontend.py:37]: @pavish This code is a little hard to follow. Do you mind if I refactor it a bit?

Comment (pavish) [reply mathesar/views/frontend.py:37]: @kgodey Sure. It's okay with me. Thank you!

**Review by kgodey — APPROVED** [review]
@pavish The functionality looks good to me. My only feedback is that I would use `currentDB` and `currentSchema` over `selectedDB"`and `selectedSchema` as variable names because I think that's a little less ambiguous. I'll leave it up to you whether to fix before merge.

I refactored the views file to be clearer, please review before merging.

**Comment by kgodey** [comment]
@pavish Also I forgot to mention in my previous comment that I changed the view behavior to return a 404 if the `db_name` or `schema_id` passed in is invalid. I think that makes more sense than returning a different schema or database.

File: mathesar/views/frontend.py
Comment (pavish) [inline mathesar/views/frontend.py:76]: @kgodey If schemas are not synced yet while request is made, or if there are no databases configured, this will result in an infinite loop of redirection leading to `too many redirects` error and the browser killing the request.

We need to handle this for edge cases where schemas may not have synced.

I understand that no db is not an immediate concern, and we need atleast one db to start mathesar at the moment. In the future, if we decide allowing mathesar to start without a db and then configure databases using the frontend, this needs to be handled.

**Comment by pavish** [comment]
> @pavish The functionality looks good to me. My only feedback is that I would use `currentDB` and `currentSchema` over `selectedDB"`and `selectedSchema` as variable names because I think that's a little less ambiguous. I'll leave it up to you whether to fix before merge.
> 
> I refactored the views file to be clearer, please review before merging.

@kgodey Makes sense. I will make this change in this PR, before merge.

File: mathesar/views/frontend.py
Comment (pavish) [reply mathesar/views/frontend.py:76]: Also all calls to `<db_name>/schemas/` would endup redirecting to the first available schema, but it should go to the schemas route, instead.

Comment (kgodey) [reply mathesar/views/frontend.py:76]: @pavish I made `schemas` a separate view, which should fix both issues.

Comment (pavish) [reply mathesar/views/frontend.py:76]: @kgodey Looks good to me! Thank you.

**Comment by pavish** [comment]
@kgodey I've made changes to use currentDB and currentSchema over selectedDB and selectedSchema, in 83c2fb49cc9b7975c24d4e0e1710fe20f1f7465e

**Review by kgodey — APPROVED** [review]
Looks good!

**Pull request merged**
